code/obj/item/tool/electronics.dm

```diff
@@ -796,97 +796,100 @@
 		else
 			boutput(user, SPAN_ALERT("No new items entered into kit."))
 
+		tgui_process.update_uis(src)
+
 	else
 		..()
 
 /obj/machinery/rkit/attack_hand(mob/user)
-	src.add_fingerprint(user)
-	var/dat
-	var/hide_allowed = src.allowed(user)
-	dat = "<b>Ruckingenur Kit</b><HR>"
-
-	dat += "<b>Scanned Items:</b><br>"
-	for(var/datum/electronics/scanned_item/S in ruck_controls.scanned_items)
-		dat += "<u>[S.name]</u><small> "
-		//dat += "<A href='?src=\ref[src];op=\ref[S];tp=done'>Frame</A>"
-		if (S.item_mats && src.olde)
-			dat += " * <A href='?src=\ref[src];op=\ref[S];tp=done'>Frame</A>"
-		else if (S.blueprint)
-			if(!S.locked || hide_allowed || src.olde)
-				dat += " * <A href='?src=\ref[src];op=\ref[S];tp=blueprint'>Blueprint</A>"
-			else
-				dat += " * Blueprint Disabled"
-		if(hide_allowed)
-			dat += " * <A href='?src=\ref[src];op=\ref[S];tp=lock'>[S.locked ? "Locked" : "Unlocked"]</A>"
-		dat += "</small><br>"
-	dat += "<br>"
-
-	dat += "<HR>"
+	src.ui_interact(user)
+
+/obj/machinery/rkit/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "RuckingenurKit", src.name)
+		ui.open()
+
+/obj/machinery/rkit/ui_data(mob/user)
+	var/list/scanned_items = list()
+
+	for (var/datum/electronics/scanned_item/item as anything in ruck_controls.scanned_items)
+		var/atom/A = item.item_type
+		scanned_items.Add(list(list(
+			name = item.name,
+			description = initial(A.desc),
+			has_item_mats = !!item.item_mats,
+			blueprint_available = !!item.blueprint,
+			locked = item.locked,
+			imagePath = getItemIcon(item.item_type, C = user.client),
+			ref = ref(item),
+		)))
+
+	. = list(
+		hide_allowed = src.allowed(user),
+		scanned_items = scanned_items,
+		olde = src.olde
```

Comment:
I have absolutely no idea what `olde` means.

Reply:
So, `/obj/machinery/rkit` have a var/olde mode, which, if enabled, makes them print electronic frames instead of blueprints. There is 1 rkit in z2.dmm with olde enabled. I made olde mode work without too much effort, but I'm also down to tear down the whole concept if I get a dev go ahead.

Reply:
as the person who did the `olde` thing, go ahead and tear it out if it makes things cleaner.
It doesn't even work too well these days anyways, since it's incompatible with anything that has a custom material list

Reply:
Maybe we can just rename the tgui property to legacyMode and we're good?

Reply:
Rename the UI prop to that, then see my other comment.